src/wast-parser.y

```diff
@@ -665,9 +665,9 @@ expr1 :
       assert(if_->type == ExprType::If);
       if_->if_.true_->label = $2;
     }
-  | try_check LPAR BLOCK labeling_opt block RPAR catch_list {
-      $5->label = $4;
+  | try_check labeling_opt LPAR BLOCK block RPAR catch_list {
```

Comment:
It is an implicit block (it doesn't build a block Expr). The block is just acting like a "then" for the "try".

Alternatively, I could make the block ALWAYS explicit. The problem is that I am trying to find a way to parse this to match the semantics of the proposal.